xmake/core/tool/builder.lua

```diff
@@ -505,21 +505,30 @@ function builder:_add_flags_from_language(flags, opt)
     for _, item in ipairs(items) do
         local check = item.check
         local mapper = item.mapper
+        local name = item.name
         local extras = item.extras
         if item.multival then
             local extra = self:_extraconf(extras, item.values)
             local results = mapper(self:_tool(), item.values, {target = target, targetkind = self:_targetkind(), extra = extra})
             for _, flag in ipairs(table.wrap(results)) do
                 if flag and flag ~= "" and (not check or self:has_flags(flag)) then
-                    table.insert(flags, flag)
+                    if name == "runtimes" then
```

Comment:
This is general logic, don't add hard-coded special handling.

you should modify order here. https://github.com/xmake-io/xmake/blob/08a71b4242c8f9f1c61bf585f6518b04e2b2de98/xmake/languages/c%2B%2B/xmake.lua#L65-L71